### PR TITLE
Entity authentication

### PR DESCRIPTION
**Pull request by TofMassilia13320**
Update entity authication support

### CODE REVIEW

File: services/stsafea/stsafea_entity_auth.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	/*- Create Rsp frame and populate elements*/
+	stse_frame_allocate(RspFrame);
+	stse_frame_element_allocate_push(&RspFrame,eRsp_header,1,&rsp_header);
+	stse_frame_element_allocate_push(&RspFrame,eChallenge,STSE_EDDSA_CHALLENGE_SIZE,pChallenge);
+
+	/*- Perform Transfer*/
+	ret = stse_frame_transfer(pSTSE,
+			&CmdFrame,
+			&RspFrame,
+			stsafea_cmd_timings[pSTSE->device_type][STSAFEA_EXTENDED_CMD_GENERATE_CHALLENGE]);
```
Comment (Vaarai) [inline services/stsafea/stsafea_entity_auth.c:56]: Please use `stsafea_extended_cmd_timings[pSTSE->device_type][STSAFEA_EXTENDED_CMD_GENERATE_CHALLENGE]` instead.

File: core/stse_platform.c [hunk 2]
```diff
@@ -64,3 +64,9 @@ __WEAK stse_ReturnCode_t stse_platform_hmac_sha256_compute(PLAT_UI8 *pSalt, PLAT
 
 	return retval;
 }
+
+__WEAK stse_ReturnCode_t stse_platform_hash_sha256_compute(PLAT_UI8 *pPayload, PLAT_UI32 payload_length,
```
[the remark this answers is not in the document]
Comment (TofMassilia13320) [reply core/stse_platform.c:68]: solved

File: services/stsafea/stsafea_asymmetric_key_slots.c [hunk 3]
```diff
@@ -371,7 +161,7 @@ stse_ReturnCode_t stsafea_generate_ecc_key_pair(
 	}
 
 	PLAT_UI8 cmd_header = STSAFEA_CMD_GENERATE_KEY;
-	PLAT_UI8 attribute_tag = STSAFEA_SUBJECT_TAG_PRIVATE_KEY_SLOT;
+	PLAT_UI8 attribute_tag = STSAFEA_SUBJECT_TAG_GENERIC_PUBLIC_KEY_SLOT;
```
[the remark this answers is not in the document]
Comment (TofMassilia13320) [reply services/stsafea/stsafea_asymmetric_key_slots.c:164]: solved

File: core/stse_generic_typedef.h [hunk 4]
```diff
@@ -456,6 +456,19 @@ typedef struct {
 	PLAT_UI16 signature_size;
 } stse_ecc_info_t;
 
+/*!
+ * \struct stsafea_generic_public_key_configuration_flags_t
+ * \brief stsafea generic public key configuration flags type
+ */
+typedef struct stsafea_generic_public_key_configuration_flags_t{
```
[the remark this answers is not in the document]
Comment (TofMassilia13320) [reply core/stse_generic_typedef.h:463]: solved

File: services/stsafea/stsafea_public_key_slots.h [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ * \details 	This service format and send/receive the get public key slots count command/response
+ * \param[in] 	pSTSE 					Pointer to STSAFE Handler
+ * \param[out] 	pPublic_key_slot_count 	Targeted STSAFE-A total public key slot count
+ * \return 		\ref stse_ReturnCode_t : STSAFE_OK on success ; error code otherwise
+ */
+stse_ReturnCode_t stsafea_query_generic_public_key_slots_count(
+		stse_Handler_t * pSTSE,
+		PLAT_UI8 * pGeneric_public_key_slot_count);
+
+stse_ReturnCode_t stsafea_query_generic_public_key_slot_info(
```
[the remark this answers is not in the document]
Comment (TofMassilia13320) [reply services/stsafea/stsafea_public_key_slots.h:66]: solved

File: services/stsafea/stsafea_entity_auth.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+	PLAT_UI8 rsp_header;
+	stse_frame_allocate(RspFrame);
+	stse_frame_element_allocate_push(&RspFrame,eRsp_header,STSAFEA_HEADER_SIZE,&rsp_header);
+	stse_frame_element_allocate_push(&RspFrame,eSignature_validity,1,pSignature_validity);
+
+	/*- Perform Transfer*/
+	ret = stse_frame_transfer(pSTSE,
+			&CmdFrame,
+			&RspFrame,
+			stsafea_cmd_timings[pSTSE->device_type][STSAFEA_CMD_VERIFY_SIGNATURE]
```
[the remark this answers is not in the document]
Comment (TofMassilia13320) [reply services/stsafea/stsafea_entity_auth.c:114]: solved

File: services/stsafea/stsafea_entity_auth.c (hunk 1 above)
Comment (TofMassilia13320) [reply services/stsafea/stsafea_entity_auth.c:56]: solved

**Review by Vaarai — APPROVED** [review]
Changes ok

**Pull request merged**